File: src/main/java/com/mobilesolutionworks/android/viper/VIPERViewBinding.java
package com.mobilesolutionworks.android.viper;

import android.databinding.ViewDataBinding;

/**
 * Created by yunarta on 5/5/17.
 */

public class VIPERViewBinding<V extends VIPERView, B extends ViewDataBinding> {

    private V view;

    private B binding;

    protected B binding() {
        return binding;
    }

    protected V view() {
        return view;
    }
}

File: src/main/kotlin/com/mobilesolutionworks/android/viper/VIPERViewBinding.kt
package com.mobilesolutionworks.android.viper

import android.databinding.ViewDataBinding

/**
 * Created by yunarta on 14/5/17.
 */

public class VIPERViewBinding<V : VIPERView, B : ViewDataBinding> {

    private val view: V? = null

    private val binding: B? = null

    protected fun binding(): B? {
        return binding
    }

    protected fun view(): V? {
        return view
    }
}
